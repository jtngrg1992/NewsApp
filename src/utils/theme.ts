export default {
  primaryColor: '#48e4c2',
  secondaryColor: '#9a9a9a',
  backgroundColor: '#fafafa',
  onPrimary: '#fff',
  onBackground: '#000',
  onBackgroundLight: '#808080',
  onSecondary: '#fff',
};
