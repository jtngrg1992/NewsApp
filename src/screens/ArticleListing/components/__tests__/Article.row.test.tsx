import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {testID} from '@news/test-setup/utils';
import mockArticles from '@news/model/articles/mock-data';
import {ArticleRow, ArticleRowProps} from '../Article.row';

describe('unit/snapshot tests for ArticleRow', () => {
  let wrapper: ShallowWrapper;

  const mockProps: ArticleRowProps = {
    onPress: jest.fn(),
    article: mockArticles[0],
  };

  beforeEach(() => (wrapper = shallow(<ArticleRow {...mockProps} />)));

  it('should render correctly with the supplied mock article', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct title in designated `Text` element', () => {
    expect(wrapper.findWhere(testID('article-title')).prop('children')).toBe(
      mockProps.article.title,
    );
  });

  it('should render correct author in designated `Text` element', () => {
    expect(wrapper.findWhere(testID('article-author')).prop('children')).toBe(
      mockProps.article.author,
    );
  });

  it('should render correct image as article thumbnail', () => {
    expect(
      wrapper.findWhere(testID('article-thumbnail')).prop('source'),
    ).toEqual({
      uri: mockProps.article.media!.thumbnail,
    });
  });

  it('should call `onPress` prop when tapped upon', () => {
    wrapper.findWhere(testID('touchable')).props().onPress();
    expect(mockProps.onPress).toBeCalledTimes(1);
  });
});
