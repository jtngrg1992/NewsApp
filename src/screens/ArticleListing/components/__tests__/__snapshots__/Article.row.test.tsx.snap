// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit/snapshot tests for ArticleRow should render correctly with the supplied mock article 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      },
      undefined,
    ]
  }
  testID="touchable"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#9a9a9a",
          "borderRadius": 30,
          "height": 60,
          "marginRight": 15,
          "overflow": "hidden",
          "width": 60,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "some.url",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
        testID="article-thumbnail"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#000",
            "fontSize": 17,
          }
        }
        testID="article-title"
      >
        Mock title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#808080",
              "fontSize": 15,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="article-author"
      >
        Mock Author
      </Text>
      <View
        style={
          Array [
            Object {
              "marginTop": 5,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#808080",
              "fontSize": 15,
            }
          }
        >
          Mock source
        </Text>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/calendar.png",
              }
            }
            style={
              Object {
                "height": 15,
                "marginRight": 6,
                "tintColor": "#808080",
                "width": 15,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#808080",
                "fontSize": 15,
              }
            }
          >
            2021-01-05
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/next.png",
          }
        }
        style={
          Object {
            "height": 15,
            "marginLeft": 15,
            "tintColor": "#808080",
            "width": 15,
          }
        }
      />
    </View>
  </View>
</ForwardRef>
`;
